tests/test-browser/chrome-test.js

```diff
@@ -0,0 +1,80 @@
+const assert = require('assert');
+const puppeteer = require('puppeteer');
+const inspect = require('util').inspect;
+
+function getUrl() {
+  return process.argv[2];
+}
+
+function testJupyterLabPage(html) {
+  if (inspect(html).indexOf('jupyter-config-data') === -1) {
+    console.error('Error loading JupyterLab page:');
+    console.error(html);
+  }
+}
+
+async function testApplication(page) {
+  const el = await page.waitForSelector('#browserTest', { timeout: 100000 });
+  console.log('Waiting for application to start...');
+  let testError = null;
+
+  try {
+    await page.waitForSelector('.completed');
+  } catch (e) {
+    testError = e;
+  }
+  const textContent = await el.getProperty('textContent');
+  const errors = JSON.parse(await textContent.jsonValue());
+
+  for (let error of errors) {
+    console.error(`Parsed an error from text content: ${error.message}`, error);
+  }
+
+  return testError;
+}
+
+function testGitExtension(html) {
+    // Test git icons are present
+    try {
+      assert(html.includes('--jp-icon-git-clone'), 'Could not find git clone icon.');
+      assert(html.includes('--jp-icon-git-pull'), 'Could not find git pull icon.');
+      assert(html.includes('--jp-icon-git-push'), 'Could not find git push icon.');
+    } catch (e) {
+      return e;
```

Comment:
Again, here what if we didn't wrap the asserts and just let the errors bubble up?